presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -1109,8 +1109,10 @@ public InsertTableHandle beginRefreshMaterializedView(Session session, TableHand
         if (catalog.isPresent()) {
             ConnectorMetadata metadata = catalog.get().getMetadata();
             ConnectorSession connectorSession = session.toConnectorSession(catalog.get().getConnectorId());
-            List<SchemaTableName> materializedViews = metadata.getReferencedMaterializedViews(connectorSession, toSchemaTableName(tableName));
-            return materializedViews.stream().map(convertFromSchemaTableName(tableName.getCatalogName())).collect(toImmutableList());
+            Optional<List<SchemaTableName>> materializedViews = metadata.getReferencedMaterializedViews(connectorSession, toSchemaTableName(tableName));
+            if (materializedViews.isPresent()) {
+                return materializedViews.get().stream().map(convertFromSchemaTableName(tableName.getCatalogName())).collect(toImmutableList());
+            }
```

Comment:
There is no need to make `getReferencedMaterializedViews` returning `Optional` right? `materializedViews.get().stream()` will still return the empty list if there is no entry.

Reply:
That's a good point, yet I think making the return type `Optional` can help us differentiate the following situations without triggering an exception.
1. It is a legit call, but just the table doesn't have any referenced materialized views => return `Optional` with an empty list
2. It is not a legit call, and something goes wrong => return `Optional.empty()`
   - The table is not an managed table
   - The table itself is a materialized view